fxgl-ai/src/test/java/com/almasb/fxgl/pathfinding/astar/AStarPathfinderTest.kt

```diff
@@ -0,0 +1,96 @@
+/*
+ * FXGL - JavaFX Game Library. The MIT License (MIT).
+ * Copyright (c) AlmasB (almaslvl@gmail.com).
+ * See LICENSE for details.
+ */
+package com.almasb.fxgl.pathfinding.astar
+
+import com.almasb.fxgl.pathfinding.CellState
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import java.util.*
+
+class AStarPathfinderTest {
+    private lateinit var grid: AStarGrid
+    private lateinit var pathfinder: AStarPathfinder
+
+    @BeforeEach
+    fun setUp() {
+        grid = AStarGrid(GRID_SIZE, GRID_SIZE)
+        pathfinder = AStarPathfinder(grid)
+    }
+
+    @Test
+    fun testFindPath() {
+        var path = pathfinder.findPath(3, 0, 5, 0)
+        assertPathEquals(path, 4, 0, 5, 0)
+
+        // Add barriers.
+        for (i in 0..4) grid[4, i].state = CellState.NOT_WALKABLE
+        path = pathfinder.findPath(3, 0, 5, 0)
+        assertPathEquals(path,
+                3, 1,
+                3, 2,
+                3, 3,
+                3, 4,
+                3, 5,
+                4, 5,
+                5, 5,
+                5, 4,
+                5, 3,
+                5, 2,
+                5, 1,
+                5, 0)
+
+        // Make passing impossible.
+        for (i in 0..19) grid[4, i].state = CellState.NOT_WALKABLE
+        path = pathfinder.findPath(3, 0, 5, 0)
+        assert(path.isEmpty())
+    }
+
+    @Test
+    fun testFindPathWithBusyCells() {
+        grid[3, 0].state = CellState.NOT_WALKABLE
+        grid[3, 1].state = CellState.NOT_WALKABLE
+        grid[3, 2].state = CellState.NOT_WALKABLE
+        grid[3, 3].state = CellState.NOT_WALKABLE
+        grid[3, 5].state = CellState.NOT_WALKABLE
+        grid[1, 4].state = CellState.NOT_WALKABLE
+        val path = pathfinder.findPath(1, 1, 4, 5, ArrayList())
+        assertPathEquals(path,
+                2, 1,
+                2, 2,
+                2, 3,
+                2, 4,
+                3, 4,
+                4, 4,
+                4, 5)
+        val pathWithBusyCell = pathfinder.findPath(1, 1, 4, 5, listOf(grid[3, 4]))
+        assertPathEquals(pathWithBusyCell,
+                2, 1,
+                2, 2,
+                2, 3,
+                2, 4,
+                2, 5,
+                2, 6,
+                3, 6,
+                4, 6,
+                4, 5)
+    }
+
+    private fun assertPathEquals(path: List<AStarCell>, vararg points: Int) {
+        val pointsList = points.toList().chunked(2) { it[0] to it[1] }
+        assert(pointsList.size == path.size) { reportNotMatchingPaths(path, pointsList) }
```

Comment:
According to [assert doc](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/assert.html) these are only active when `-ea` JVM option is enabled.

If the above is true, please replace all `assert` with `assertEquals`, which can also store a failure message.

Reply:
Is it alright to use the Kotlin `assertEquals`? Not sure I added the dependency correctly.

Reply:
![image](https://user-images.githubusercontent.com/9862282/94891082-6105e280-0481-11eb-9fd1-d8c9b05c1643.png)
You don't know by chance why I get this warning? I guess my setup is somehow wrong.
Anyway, running works fine so there is no issue.

Reply:
It's an IDE bug, I've also got it, just ignore it.